src/Controller/Frontend.php

```diff
@@ -62,8 +62,8 @@ public function before(Request $request)
         if ($this->getOption('general/maintenance_mode')) {
             if (!$this->isAllowed('maintenance-mode')) {
                 $template = $this->templateChooser()->maintenance();
-                $response = $this->render($template);
-                $response->setStatusCode(Response::HTTP_SERVICE_UNAVAILABLE);
+                $html = $this->app['twig']->render(reset($template));
```

Comment:
If the theme specific template doesn't exist there's no fallback to the global one now. Call twig->resolveTemplate->render().

Also you could still use a TemplateResponse here.